File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    elements += elements
    ans = set()
    dp = [[0]*(2*n) for _ in range(n+1)]
    for i in range(2*n):
        dp[1][i] = elements[i]
        ans.add(dp[1][i])
    for l in range(2,n+1):
        for i in range(2*n-l):
            dp[l][i] = dp[l-1][i] + dp[l-1][i+1] - dp[l-2][i+1]
            ans.add(dp[l][i])
    return len(ans)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.kt
class Solution {
    fun solution(elements: IntArray): Int {
        val ans = mutableSetOf<Int>()
        val n = elements.size
        for (i in 0 until n) {
            var e = elements[i]
            ans.add(e)
            for (j in i+1 until i+n) {
                e += elements[j%n]
                ans.add(e)
            }
        }
        return ans.size
    }
}